fcomm/src/lib.rs

```diff
@@ -1085,6 +1086,48 @@ mod test {
     use lurk::public_parameters::public_params;
     use lurk::z_data::{from_z_data, to_z_data};
 
+    // You have broken a snapshot test. Unlike round-trip tests, those tests check the actual format of serialized Lurk expressions,
+    // and since you broke one, it's probable that you have changed that format, which will break at least the fcomm examples.
+    // Please read the documentation on snapshot tests `https://insta.rs/docs/quickstart/`, fix the snapshot **AND**
+    // make sure `cargo nextest run --run-ignored all -E 'test(test_make_fcomm_examples)'` passes
+    #[test]
+    fn test_snapshot_serialized_expressions() {
+        let function_source: &str = "(letrec ((secret 12345) (a (lambda (acc x) (let ((acc (+ acc x))) (cons acc (hide secret (a acc))))))) (a 0))";
+        let function_inputs: &str = "(+ 1 2)";
+        let committed_expression = CommittedExpression::<S1> {
+            expr: LurkPtr::Source(function_source.into()),
+            secret: None,
+            commitment: None,
+        };
+        assert_json_snapshot!(committed_expression);
+
+        let input = Expression::<S1> {
+            expr: LurkPtr::Source(function_inputs.into()),
+        };
+        assert_json_snapshot!(input);
+
+        let c = Commitment {
+            comm: S1::from(123),
+        };
+        assert_json_snapshot!(c);
+
+        let req = OpeningRequest {
+            input,
+            commitment: c.clone(),
+            chain: true,
+        };
+        assert_json_snapshot!(req);
```

Comment:
This is just going to error, since the commitment cannot possibly be opened. Even if it could and corresponded to the function whose source we snapshotted above, the input is malformed for the function.

Reply:
Yes. I consider this to be out of scope.